src/interactive.cr

```diff
@@ -0,0 +1,114 @@
+require "colorize"
+require "./game"
+
+abstract class StdInReader
+  def gets
+  end
+end
+
+class CoreStdInReader < StdInReader
+  def gets
+    STDIN.gets
+  end
+end
+
+class Interactive
+  # exposed for replacement for tests
+  property stdin : StdInReader = CoreStdInReader.new
+  getter game : Game
+
+  def initialize(game : Game = Game.new)
+    # Assume that if no moves have been played that the game's board
+    # should be setup with all the pieces needed for a new game
+    if game.moves.size == 0
+      game.setup_board
+    end
+
+    @game = game
+  end
+
+  def move_input : GameTrackedMove?
+    puts "Enter a move by specifying both coordinates separated by a space (eg: \"a2 a3\"):"
+    from_stdin = @stdin.gets
+    puts "\n"
+
+    if from_stdin.nil?
+      return nil
+    end
+
+    coordinates = from_stdin.chomp.split(" ")
+    if coordinates.size != 2
+      puts "Invalid input, specify both coordinates separated by a space (eg: \"a2 a3\"):"
+      nil
+    end
+
+    board_coordinates : Array(BoardCoordinate) = coordinates.map do |coordinate|
+      invalid_coordinate_message = "Invalid coordinate #{coordinate}. Coordinate should be a letter and a number, like \"a2\""
+      if coordinate.size != 2
+        puts invalid_coordinate_message
+        return nil
+      end
+
+      rank_and_file = coordinate.split("")
+      board_movement = {rank_and_file[0][0], rank_and_file[1].to_i}
+
+      # this tests whether or not the board_movement is valid and raises if not,
+      # and should likely be replaced with a validity check
+      convert_board_coordinate_to_array_matrix_coordinate(board_movement)
+
+      board_movement
+    end
+
+    {from: board_coordinates[0], to: board_coordinates[1]}
+  end
+
+  def game_header
+    white_circle = "●"
+    black_circle = "○"
+    current_turn_circle = @game.turn.to_sym == :white ? white_circle : black_circle
+
+    game_header = <<-GAME_HEADER
+    #{"Current turn".colorize.bright}
+      #{current_turn_circle} #{@game.turn.to_s.downcase}
```

Comment:
Ensure that the `current_turn_circle` is colorized with a foreground of white and a background of black to override any terminals that might be black text on white